test/Microsoft.AspNetCore.Http.Abstractions.Tests/HostStringTest.cs

```diff
@@ -123,5 +124,43 @@ public void NotEquals_EmptyHostStringAndNonNullHostString()
             // Act and Assert
             Assert.NotEqual(hostString, new HostString(string.Empty));
         }
+
+        [Theory]
+        [InlineData("localHost", "localhost")]
+        [InlineData("localHost", "*")] // Any - Used by HttpSys
+        [InlineData("localhost:9090", "localHost")]
+        [InlineData("example.com:443", "example.com")]
+        [InlineData("foo.eXample.com:443", "*.exampLe.com")]
+        [InlineData("f.eXample.com:443", "*.exampLe.com")]
```

Comment:
Any test case to verify that "a.b.c.example.com" matches the "*.example.com" filter?